packages/test-factory/src/factory.test.ts

```diff
@@ -878,4 +878,25 @@ describe('factory', () => {
       expect(factory.user.build().data).toBe(data);
     });
   });
+
+  describe('pick with create (no argument)', () => {
+    it('should only generate values for picked columns in the async branch', async () => {
+      // This test verifies the fix for a bug where the else branch in processCreateData
+      // iterated over `fns` (all factory functions) instead of `pick` (filtered keys).
+      // The bug caused identity columns to be generated even when they should be
+      // auto-generated by the database.
+      
+      const pickedFactory = factory.user.pick({ name: true, age: true });
+      
+      // When calling create() without arguments, only picked columns should be generated.
+      // The identity column (id) should NOT be in the generated data before insert.
+      const user = await pickedFactory.create();
+      
+      // The user should have an id (from database), but only name and age should
+      // have been generated by the factory
+      expect(user.id).toBeDefined();
```

Comment:
For me TS highlights this line, wondering how it didn't work for you.

Since the `id` is not in the pick list, why do you expect it to be defined?